pkg/api/v1alpha1/cluster_types.go

```diff
@@ -216,7 +218,18 @@ func WorkerNodeGroupConfigurationsSliceEqual(a, b []WorkerNodeGroupConfiguration
 			delete(m, k)
 		}
 	}
-	return len(m) == 0
+	if len(m) != 0 {
+		return false
+	}
+
+	for index, slice := range a {
+		sameTaints := TaintsSliceEqual(slice.Taints, b[index].Taints)
```

Comment:
For now this will work as we have only one node group. However when the multiple node group feature is added, we can do this by indexing the worker node groups with their names. Will add that as a work item.

Reply:
This would just need to be worked on then when the multiple worker node group feature is.

Reply:
I'm ok with doing it later :)
@bnrjee pls post here the issue when you create it and feel free to resolve this comment (unless @vivek-koppuru thinks differently)

Reply:
I have not created an issue, because this code is not checked in. So I have nothing to create an issue against. But I have added as an action item in the pr itself.
https://github.com/aws/eks-anywhere/issues/840

Reply:
I'm not sure why we can't create an issue?
Also I'm not sure which of those action items is referring to this conversation

Reply:
captured AI in https://github.com/aws/eks-anywhere/pull/977